crates/dprint/src/paths.rs

```diff
@@ -80,3 +94,27 @@ pub async fn get_and_resolve_file_paths(config: &ResolvedConfig, args: &FilePatt
   // Eventually this could should maybe be changed to use tokio tasks
   tokio::task::spawn_blocking(move || glob(&environment, &base_dir, file_patterns)).await.unwrap()
 }
+
+fn get_plugin_patterns(plugins: &[Box<dyn Plugin>]) -> Vec<String> {
+  let mut result = Vec::new();
+  let mut file_names = HashSet::new();
+  let mut file_exts = HashSet::new();
+  for plugin in plugins {
+    file_names.extend(plugin.file_names());
+    file_exts.extend(plugin.file_extensions());
+  }
+  if !file_exts.is_empty() {
+    result.push(format!("**/*.{{{}}}", file_exts.into_iter().map(|s| s.as_str()).collect::<Vec<_>>().join(",")));
+  }
+  if !file_names.is_empty() {
+    result.push(format!("**/{{{}}}", file_names.into_iter().map(|s| s.as_str()).collect::<Vec<_>>().join(",")));
+  }
+
+  // add the associations last as they are least likely to be matched
+  for plugin in plugins {
+    if let Some(associations) = plugin.get_config().0.associations.as_ref() {
+      result.extend(process_config_patterns(associations));
```

Comment:
This is wrong.

1. If there are positive association globs, then they take over completely from the file names and extensions.
2. If there are negative association globs, then they should be filtered out and not affect the file path globbing.

I should actually look into assigning paths to plugins at the same time as globbing to avoid mistakes and duplicated logic, though it might not be worth it.